src/main/java/seedu/address/model/person/Donor.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.model.person;
+
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Donor in the address book.
+ * Inherits from Person and includes donated amount from the donor.
+ * Guarantees: details are present and not null, field values are validated.
+ */
+public class Donor extends Person {
+
+    private final DonatedAmount donatedAmount;
+
+    /**
+     * Constructs a {@code Donor}.
+     *
+     * @param name    Name of the donor.
+     * @param phone   Phone number of the donor.
+     * @param email   Email address of the donor.
+     * @param address Address of the donor.
+     * @param tags    Tags associated with the donor.
+     * @param donatedAmount   Hours contributed by the donor.
+     */
+    public Donor(Name name, Phone phone, Email email, Address address, Set<Tag> tags, DonatedAmount donatedAmount) {
+        super(name, phone, email, address, tags);
+        requireAllNonNull(donatedAmount);
+        this.donatedAmount = donatedAmount;
+    }
+
+    /**
+     * Compares this Donor to another for equality.
+     *
+     * @param other The object to compare with.
+     * @return True if both Donors have identical attributes.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof seedu.address.model.person.Donor)) {
```

Comment:
Should this be just `Donor` instead of `seedu.address.model.person.Donor`

Or is there a reason for the longer name